src/reanimated2/hook/useScrollViewOffset.ts

```diff
@@ -15,23 +21,49 @@ const scrollEventNames = [
   'onMomentumScrollEnd',
 ];
 
+const addListenerToScroll = (
+  offsetRef: ScrollSharedValue<number>,
+  animatedRef: any
+) => {
+  runOnUI(() => {
+    'worklet';
+    offsetRef.triggerScrollListener = true;
+    offsetRef.addListener(animatedRef(), (newValue: any) => {
+      if (offsetRef.triggerScrollListener) {
+        scrollTo(animatedRef, 0, Number(newValue), false);
+      }
+    });
+  })();
+};
+
 export function useScrollViewOffset(
   aref: RefObject<Animated.ScrollView>
 ): SharedValue<number> {
-  const offsetRef = useRef(useSharedValue(0));
+  const offsetRef: ScrollSharedValue<number> = useRef(
```

Comment:
I know this was here before but i wonder why is useSharedValue wrapped with useRef? This does not look like a typicaly pattern and if it in fact is necessary than it'd be good to add some comment as to why is it designed this way. I think just using `useSharedValue` here should suffice